pkg/project/serverlessoperator/project.go

```diff
@@ -0,0 +1,58 @@
+package serverlessoperator
+
+import (
+	"fmt"
+	"os"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+// Metadata is the structure for project metadata
+// of SOs project.yaml
+type Metadata struct {
+	Project      Project      `yaml:"project"`
+	Olm          Olm          `yaml:"olm"`
+	Requirements Requirements `yaml:"requirements"`
+}
+
+type Project struct {
+	Name    string `yaml:"name"`
+	Version string `yaml:"version"`
+}
+
+type Olm struct {
+	Replaces  string   `yaml:"replaces"`
+	SkipRange string   `yaml:"skipRange"`
+	Channels  Channels `yaml:"channels"`
+}
+
+type Requirements struct {
+	Kube       Kube       `yaml:"kube"`
+	Golang     string     `yaml:"golang"`
+	Nodejs     string     `yaml:"nodejs"`
+	OcpVersion OcpVersion `yaml:"ocpVersion"`
+}
+
+type Channels struct {
+	Default string   `yaml:"default"`
+	List    []string `yaml:"list"`
+}
+
+type Kube struct {
+	MinVersion string `yaml:"minVersion"`
+}
+
+type OcpVersion struct {
+	Min   string `yaml:"min"`
+	Max   string `yaml:"max"`
+	Label string `yaml:"label"`
+}
```

Comment:
can't we reuse the Metadata struct we already have?

Also I don't think we need all these fields?

Reply:
I thought the existing Metadata struct was for a different purpose / structure of project.yaml. But if you're OK to reuse it for this, I am fine too. Will update it...